### PR TITLE
feat(data-table): Added .destroy() method to data-table component

### PR DESCRIPTION
**Pull request by abhiomkar**
#### Description

* Added `.destroy()` method to data table component.
* Remove overriding touch dimension of row checkbox.
* Removed testing with mocks in data table component test.

Refs #4591

### CODE REVIEW

File: packages/mdc-data-table/component.ts [hunk 1]
```diff
@@ -80,6 +88,14 @@ export class MDCDataTable extends MDCComponent<MDCDataTableFoundation> {
     this.foundation_.setSelectedRowIds(rowIds);
   }
 
+  destroy() {
+    this.headerRow_.removeEventListener('change', this.handleHeaderRowCheckboxChange_);
+    this.content_.removeEventListener('change', this.handleRowCheckboxChange_);
+
+    this.headerRowCheckbox_.destroy();
```
Comment (moog16) [inline packages/mdc-data-table/component.ts]: I think you need an `if (this.headerRowCheckbox_)` check

File: test/screenshot/spec/mdc-data-table/classes/baseline-checkbox.html [hunk 2]
```diff
@@ -97,7 +97,7 @@
             </tr>
             <tr data-row-id="u2" class="mdc-data-table__row mdc-data-table__row--selected" aria-selected="true">
               <td class="mdc-data-table__cell mdc-data-table__cell--checkbox">
-                <div class="mdc-checkbox mdc-data-table__row-checkbox">
+                <div class="mdc-checkbox mdc-data-table__row-checkbox mdc-checkbox--selected">
```
Comment (moog16) [inline test/screenshot/spec/mdc-data-table/classes/baseline-checkbox.html]: nit: this row is "Offline". Is the Online/Offline & checkbox supposed to be related?

File: test/unit/mdc-data-table/mdc-data-table.test.js [hunk 3]
```diff
@@ -279,76 +255,13 @@ test('adapter#setAttributeAtRowIndex', () => {
...
-    component,
-    adapter,
-  } = setupTestWithMocks();
-
-  const rows = [].slice.call(root.querySelectorAll(strings.ROW_SELECTOR));
-  td.when(mockFoundation.getRows()).thenReturn(rows);
-
-  adapter.registerRowCheckboxes();
-
-  td.verify(mockCheckboxFactory(td.matchers.isA(Element)), {times: 3});
```
Comment (moog16) [inline test/unit/mdc-data-table/mdc-data-table.test.js]: isn't this test still valid with the exception of the component.destroy() call?

File: packages/mdc-data-table/component.ts (hunk 1 above)
Comment (abhiomkar) [reply packages/mdc-data-table/component.ts]: As per current implementation `headerRowCheckbox_` is non-nullable and guaranteed to be initialized in `initialSyncWithDOM` method.

File: test/screenshot/spec/mdc-data-table/classes/baseline-checkbox.html (hunk 2 above)
Comment (abhiomkar) [reply test/screenshot/spec/mdc-data-table/classes/baseline-checkbox.html]: They're not related. Checkbox selects the table row.

See this for example: https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2019/06/04/16_18_48_988/spec/mdc-data-table/classes/baseline-checkbox.html?utm_source=golden_json

File: test/unit/mdc-data-table/mdc-data-table.test.js (hunk 3 above)
Comment (abhiomkar) [reply test/unit/mdc-data-table/mdc-data-table.test.js]: It is a valid test but mocking isn't helping much. But, I discovered more bugs without mocking compared to with mocking.

**Pull request merged**